packages/components/src/types/common.ts

```diff
@@ -0,0 +1,13 @@
+export type TextWithA11y = {
+  /** Text to display */
+  text: string
+  /** Accessibility override for text */
+  a11y?: string
+}
+
+export type TextWithA11yAndValue = TextWithA11y & {
+  /** Value that may differ from provided text */
+  value?: string | number
+}
```

Comment:
Perhaps tied to my other comment on not really understanding this property, but at face value it seems like this should live with the CheckboxGroup and perhaps move to `FormText.tsx` for the Radio button component when it's actually shared. Outside those use cases, it's unclear to me what purpose this would have as a shared typing across most components.